include/pybind11/embed.h

```diff
@@ -261,6 +283,13 @@ class scoped_interpreter {
         initialize_interpreter(init_signal_handlers, argc, argv, add_program_dir_to_path);
     }
 
+#if PY_VERSION_HEX >= PYBIND11_PYCONFIG_SUPPORT_PY_VERSION
+    explicit scoped_interpreter(const PyConfig &config, bool add_program_dir_to_path = true) {
```

Comment:
I think this needs to become:
```
    explicit scoped_interpreter(PyConfig *config,
                                int argc = 0,
                                const char *const *argv = nullptr,
                                bool add_program_dir_to_path = true) {
````

I'm not sure how this will shake out best.

I'm thinking the `scoped_config` will not actually be needed.

I'd definitely split out `initialize_interpreter_pre_pyconfig()` for clarity and safety (unused variables in one branch or another), in the `detail` namespace.

Then there would be two `initialize_interpreter()` overloads:

```
#if PY_VERSION_HEX >= PYBIND11_PYCONFIG_SUPPORT_PY_VERSION_HEX
inline void initialize_interpreter(PyConfig *config,
                                   int argc = 0,
                                   const char *const *argv = nullptr,
                                   bool add_program_dir_to_path = true) {
    ... implementation here ...
}
#endif

inline void initialize_interpreter(bool init_signal_handlers = true,
                                   int argc = 0,
                                   const char *const *argv = nullptr,
                                   bool add_program_dir_to_path = true) {
#if PY_VERSION_HEX < PYBIND11_PYCONFIG_SUPPORT_PY_VERSION_HEX
    detail::initialize_interpreter_pre_pyconfig(
        init_signal_handlers, argc, argv, add_program_dir_to_path);
#else
    PyConfig config;
    PyConfig_InitIsolatedConfig(&config);
    config.isolated = 0;
    config.use_environment = 1;
    config.install_signal_handlers = init_signal_handlers ? 1 : 0;
    initialize_interpreter(&config, argc, argv, add_program_dir_to_path);
#endif
}
```

Then the old `scoped_interpreter` constructor can stay as-is, and the new one will simply be:

```
    explicit scoped_interpreter(PyConfig *config,
                                int argc = 0,
                                const char *const *argv = nullptr,
                                bool add_program_dir_to_path = true) {
        initialize_interpreter(config, argc, argv, add_program_dir_to_path);
    }
```

Note that I'd choose to make this `PyConfig *config` instead of `PyConfig &config`, so that it is more obvious that the `config` is changed in place (`PyConfig_Clear()`). It seems very unlikely that anyone would want `config` to be `const`, although that's only a best guess. When you're done with this PR (CI passes), we should run it by Victor Stinner (not tagging him just now intentionally).